pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -442,6 +442,15 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 
 	// Check for any conditions that might require scheduling retry
 
+	// When pod is removed from scheduling queue because of deletion or any
+	// other reasons, binding operation should be cancelled. There is no need
+	// to check PV/PVC bindings any more.
+	// We check pod binding cache here which will be cleared when pod is
+	// removed from scheduling queue.
+	if b.podBindingCache.GetDecisions(pod) == nil {
```

Comment:
Can you add a unit test for this, if one doesn't already exist?

Reply:
sure, it has some conflicts with #70026, I will rebase and squash after #70026 is merged.

Reply:
There is [an existing test `pod-deleted-after-time`](https://github.com/kubernetes/kubernetes/pull/72045/files#diff-7e773cadec5750de298ed8fa194edda3L1357). In old PR, BindPodVolume failed with timeout error, so it passed too. Perhaps it's better to check error too. I can address this in a separate PR.